rs/artifact_pool/src/ingress_pool.rs

```diff
@@ -375,9 +375,22 @@ impl MutablePool<SignedIngress> for IngressPoolImpl {
                 }
             }
         }
+
+        let mut mutations = Vec::with_capacity(purged.len());
+        mutations.extend(
+            artifacts_with_opt
+                .drain(..)
+                .filter(|x| {
+                    !purged.contains(&IngressMessageId::new(
+                        x.artifact.expiry_time(),
+                        x.artifact.id(),
+                    ))
+                })
+                .map(ArtifactMutation::Insert),
+        );
+        mutations.extend(purged.drain().map(ArtifactMutation::Remove));
```

Comment:
Given that we now have to do this for every pool, are we sure that what we had before this PR wasn't the better abstraction?

Reply:
I personally like the current abstraction (in this PR) better than what we had before. 

Just thinking about how mutations are chained and written on a transaction in DB APIs - the order matters. So in a way we do it in a more idiomatic way with this MR.

E.g. spanner mutations - https://cloud.google.com/spanner/docs/reference/rpc/google.spanner.v1#google.spanner.v1.CommitRequest

Reply:
@eichhorl does it make sense?

Reply:
> Just thinking about how mutations are chained and written on a transaction in DB APIs - the order matters.

That makes sense, and I guess the same applies to our pools. Should we return the `ArtifactMutation`s in the same order that we applied the corresponding `ChangeAction`s to the pools?

Reply:
totally!